pkg/operator/ceph/cluster/controller.go

```diff
@@ -341,17 +344,46 @@ func (c *ClusterController) onK8sNodeUpdate(oldObj, newObj interface{}) {
 		logger.Warningf("Expected Node but handler received %#v", oldObj)
 		return
 	}
+	newNodeSchedulable := k8sutil.GetNodeSchedulable(*newNode)
+	oldNodeSchedulable := k8sutil.GetNodeSchedulable(*oldNode)
+	clusters := c.getTenantClusters(newNode)
 
-	if k8sutil.GetNodeSchedulable(*newNode) == false {
-		logger.Debugf("Skipping cluster update. Updated node %s is unschedulable", newNode.Labels[apis.LabelHostname])
-		return
+	for _, cluster := range clusters {
+		logger.Debugf("NodeUpdate: oldNodeSchedulable: %v, newNodeSchedulable: %v", oldNodeSchedulable, newNodeSchedulable)
+		if cluster.Info.IsInitialized() {
+			nodes, err := GetAllStorageNodes(cluster.context, cluster.Namespace)
+			if err != nil {
+				logger.Errorf("Error getting all storage nodes for cluster: %v", err)
+			}
+			allSchedulable := true
+			for _, node := range nodes {
```

Comment:
How often is this event of updating a node going to be raised? I had the impression that this event could be noisy, at least for a large cluster. If these events are very often, we may need to find a way to batch processing these events.

Reply:
I'll look into creating a nodeMonitor goroutine that batches events.

Reply:
Hoping to make this my next PR